File: WordSwell.Frontend/src/Faculty/Backend/BoardCont/PostViewResultModel.ts
import { BoardPost } from '@/Faculty/Backend/ModelsDB/Board/BoardPost';
import { BoardPostContents } from '@/Faculty/Backend/ModelsDB/Board/BoardPostContents';
import { FileDbInfo } from '@/Faculty/Backend/ModelsDB/FileDb/FileDbInfo';

/** 지정한 게시판의 지정한 게시물을 보기 결과 리스트 */
export interface PostViewResultModel 
{
    /** 게시물 */
    Post?: BoardPost,
    /** 게시물의 내용물 */
    PostContents?: BoardPostContents,
    /** 첨부 파일 리스트 */
    FileList?: FileDbInfo[],
    InfoCode: string,
    Message: string,
}

File: WordSwell.ApiModels/BoardCont/PostViewResultModel.cs
using DGUtility.ApiResult;
using ModelsDB.Board;
using ModelsDB.FileDb;
using WordSwell.ApiModels.FileDb;

namespace WordSwell.ApiModels.BoardCont;

/// <summary>
/// 지정한 게시판의 지정한 게시물을 보기 결과 리스트
/// </summary>
public class PostViewResultModel : ApiResultBaseModel
{
    /// <summary>
    /// 게시물
    /// </summary>
    public BoardPost? Post { get; set; }

    /// <summary>
    /// 게시물의 내용물
    /// </summary>
    public BoardPostContents? PostContents { get; set; }

    /// <summary>
    /// 첨부 파일 리스트
    /// </summary>
    public List<FileDbInfo>? FileList { get; set; }
}
